# Mutual Funds Analysis

## Business Problem

Financial advisors and retail investors often need to compare thousands of mutual funds with different performance histories, risk levels, fees, and investment requirements. The volume and complexity of available information make it difficult to evaluate funds efficiently and consistently. A structured data analysis can uncover relationships among key fund characteristics and generate insights that support more informed investment decisions.

## Project Objective

Analyze a comprehensive mutual funds dataset using Python and Pandas to explore fund characteristics, compare performance, and generate data-driven insights that support more informed investment decisions.

## Initial Business Questions

1. How do mutual fund performance metrics vary across different funds and investment categories?
2. What relationships exist between fund characteristics and historical performance?
3. How do risk-related measures compare with historical fund performance?
4. How do investment requirements differ across mutual funds?
5. What insights can help financial advisors and retail investors compare mutual funds more effectively?

---

### Tools

- Python
- Pandas
- Jupyter Notebook

### Dataset

This project uses a publicly available mutual funds dataset containing information on 23,783 mutual funds and 298 variables, including fund characteristics, historical performance, risk measures, investment requirements, and other financial metrics.

In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv("../data/MutualFunds.csv")

In [2]:
df.head()

,fund_symbol,quote_type,region,fund_short_name,fund_long_name,currency,initial_investment,subsequent_investment,fund_category,fund_family,...,peer_environment_avg,peer_environment_max,social_score,peer_social_min,peer_social_avg,peer_social_max,governance_score,peer_governance_min,peer_governance_avg,peer_governance_max
0,AAAAX,MutualFund,US,DWS RREEF Real Assets Fund - Cl,DWS RREEF Real Assets Fund - Class A,USD,1000.0,50.0,World Allocation,DWS,...,5.05,10.58,7.43,5.98,9.07,11.30,5.43,4.26,7.14,8.11
1,AAAEX,MutualFund,US,AllianzGI Health Sciences Fund,Virtus AllianzGI Health Sciences Fund Class P,USD,1000000.0,NaN,Health,Virtus,...,1.43,3.27,12.96,9.52,12.87,15.08,8.40,4.96,7.68,10.30
2,AAAFX,MutualFund,US,NaN,American Century One Choice Blend+ 2015 Portfo...,USD,2500.0,50.0,Target-Date 2015,American Century Investments,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,AAAGX,MutualFund,US,Thrivent Large Cap Growth Fund,Thrivent Large Cap Growth Fund Class A,USD,2000.0,50.0,Large Growth,Thrivent Funds,...,2.70,5.81,10.13,7.25,10.14,11.97,8.03,5.30,7.54,8.90
4,AAAHX,MutualFund,US,NaN,American Century One Choice Blend+ 2015 Portfo...,USD,5000000.0,NaN,Target-Date 2015,American Century Investments,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [3]:
df.shape

(23783, 298)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23783 entries, 0 to 23782
Columns: 298 entries, fund_symbol to peer_governance_max
dtypes: float64(277), object(21)
memory usage: 54.1+ MB


In [5]:
df.describe()

,initial_investment,subsequent_investment,total_net_assets,year_to_date_return,day50_moving_average,day200_moving_average,week52_high_low_change,week52_high_low_change_perc,week52_high,week52_high_change,...,peer_environment_avg,peer_environment_max,social_score,peer_social_min,peer_social_avg,peer_social_max,governance_score,peer_governance_min,peer_governance_avg,peer_governance_max
count,1.648500e+04,8943.000000,2.374900e+04,23382.000000,23782.000000,23782.000000,23496.000000,23496.000000,23496.000000,23421.000000,...,14839.000000,14839.000000,15388.000000,14839.000000,14839.000000,14839.000000,15388.000000,14839.000000,14839.000000,14839.000000
mean,5.179181e+06,1015.925752,4.924192e+09,0.093485,23.638731,23.084989,4.643490,0.146057,24.995830,-1.957576,...,4.357856,7.675547,8.043626,6.906983,9.346850,11.308239,6.210819,5.387539,7.219275,8.917380
std,9.253584e+07,11023.969484,2.109765e+10,0.082276,33.126315,31.552946,9.312633,0.092420,35.208749,4.069716,...,1.498273,2.838513,3.674170,1.167251,1.204409,1.732701,2.800713,0.748176,0.820705,1.364138
min,1.000000e+01,1.000000,1.000000e+01,-0.522800,1.279000,1.363000,0.000000,0.000000,1.700000,-101.420040,...,0.750000,0.830000,0.000000,2.300000,4.410000,5.860000,0.000000,2.610000,3.040000,3.310000
25%,1.000000e+03,50.000000,1.445834e+08,0.026100,10.997250,10.968000,0.750000,0.061890,11.350000,-1.960000,...,3.810000,5.810000,8.240000,6.530000,9.060000,10.710000,6.230000,5.050000,6.930000,7.950000
50%,2.500000e+03,50.000000,6.443283e+08,0.092000,14.959500,14.709000,2.420000,0.149255,15.660000,-0.800000,...,4.080000,7.420000,9.480000,7.040000,9.450000,11.520000,7.350000,5.300000,7.420000,8.950000
75%,5.000000e+05,100.000000,2.453409e+09,0.149700,25.184500,24.693250,5.390000,0.211635,26.680000,-0.340000,...,5.000000,9.890000,10.190000,7.420000,10.120000,12.040000,7.740000,5.830000,7.540000,9.650000
max,5.000000e+09,500000.000000,7.534100e+11,0.578900,2018.338000,1920.319000,455.750000,0.882960,2092.820000,0.004900,...,14.350000,21.290000,15.080000,14.780000,14.790000,17.250000,14.020000,11.650000,11.650000,15.300000


In [6]:
df["initial_investment"].describe()

count    1.648500e+04
mean     5.179181e+06
std      9.253584e+07
min      1.000000e+01
25%      1.000000e+03
50%      2.500000e+03
75%      5.000000e+05
max      5.000000e+09
Name: initial_investment, dtype: float64

## Initial Dataset Exploration - Observations

### Dataset Overview
- The dataset contains 23,783 mutual fund records and 298 variables.
- Most variables are numerical (277 float columns), while 21 are categorical/text columns.

### Data Quality
- The `initial_investment` column contains missing values and will require further investigation.
- Additional missing-value analysis will be performed across the full dataset.

### Initial Findings
- The `initial_investment` distribution appears to be highly right-skewed.
- The mean initial investment (~$5.18M) is much higher than the median ($2,500), suggesting that a relatively small number of funds with very large investment requirements may be influencing the average.

### Questions for Further Analysis
- What exactly does the `initial_investment` column represent?
- Which funds have the highest initial investment requirements?
- Are the extreme values legitimate or potential outliers?
- How many missing values exist in each column?

In [7]:
df.isnull().sum()

fund_symbol               0
quote_type                0
region                    0
fund_short_name         824
fund_long_name            5
                       ... 
peer_social_max        8944
governance_score       8395
peer_governance_min    8944
peer_governance_avg    8944
peer_governance_max    8944
Length: 298, dtype: int64

In [8]:
missing_values = df.isnull().sum().sort_values(ascending=False)
missing_values.head(15)

last_cap_gain                    23776
fund_return_2021_q3              23721
fund_max_deferred_sales_load     20780
fund_return_2021_q2              20166
fund_max_front_end_sales_load    20090
fund_return_2000_q4              19827
fund_return_2000                 19827
fund_return_2000_q3              19734
fund_return_2000_q2              19599
fund_return_2000_q1              19462
fund_return_2001                 19278
fund_return_2001_q4              19278
fund_return_2001_q3              19051
fund_return_2001_q2              18926
fund_return_2001_q1              18797
dtype: int64

## Missing Values - Observations

- Several variables contain missing values, but the extent of missingness varies considerably across the dataset.
- Some variables, such as `last_cap_gain`, are missing values for nearly all observations and may have limited analytical value.
- Historical return variables from the early 2000s also contain many missing values. A possible explanation is that many mutual funds in the dataset were created after those years and therefore do not have historical return data for that period.
- Additional investigation is required before deciding whether any variables should be removed or excluded from specific analyses.